### PR TITLE
Flippchen exploration

### PR DESCRIPTION
**Pull request by Flippchen**
- Added Data exploration
- Added first model and audio_to_image preprocessing
- Added preprocessing folder structure
- Added first Version of data split
- Added GitHub Workflow

I have added the split of test,train, val here to GitHub until we then use ClearML together

### CODE REVIEW

File: src/preprocess/dataset_split/data_subset.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import os
+import pandas as pd
+from sklearn.model_selection import GroupShuffleSplit
```
Comment (anwielts) [inline src/preprocess/dataset_split/data_subset.py:3]: I think we should use the StratifiedShuffleSplit here. GroupShuffleSplit does not guarantee that the class distribution is preserved in the splits

File: requirements.txt
Comment (anwielts) [inline requirements.txt:1]: Here the versions of e.g. torch and librosa are missing

File: src/preprocess/adio_to_image/__init__.py
Comment (anwielts) [inline src/preprocess/adio_to_image/__init__.py:1]: The folder name has a typo: adio_to_image ;)

File: src/preprocess/adio_to_image/preprocess.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    waveform = load_audio(file_path)
+    mel_spectrogram = get_mel_spectrogram(waveform)
+
+    # Apply augmentations
+    mel_spectrogram = apply_augmentations(mel_spectrogram)
+
+    return mel_spectrogram
+
+
+def plot_mel_spectrogram(mel_spectrogram, title='Mel Spectrogram', cmap='viridis', dynamic_range=80):
```
Comment (anwielts) [inline src/preprocess/adio_to_image/preprocess.py]: I would seperate the plotting functionalities into a different file or even a different directory (e.g. src/plotting/YOUR_FILE_NAME.py)

File: src/model_audio_to_video.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+warnings.filterwarnings("ignore")
+
+
+def load_metadata(file_path):
+    df = pd.read_csv(file_path)
+    num_classes = df['primary_label'].nunique()
+    return df, num_classes
+
+
+class CustomCNN(nn.Module):
```
Comment (anwielts) [inline src/model_audio_to_video.py]: Just an idea: The whole model definition could also be seperated into a different audio_to_image_model_def.py file.

File: src/model_audio_to_video.py
Comment (anwielts) [inline src/model_audio_to_video.py:1]: Why is it named audio to **video**?

**Review by anwielts — COMMENTED** [review]
Overall just some naming comments and hwo to seperate files to get to more concise single python files. The stratified splittin is something you (we) should have a look at

File: src/model_audio_to_video.py
Comment (Flippchen) [reply src/model_audio_to_video.py:1]: I named it wrong

File: src/model_audio_to_video.py (hunk 3 above)
Comment (Flippchen) [reply src/model_audio_to_video.py]: Good Idea. I also made it to its own module

File: src/preprocess/adio_to_image/preprocess.py (hunk 2 above)
Comment (Flippchen) [reply src/preprocess/adio_to_image/preprocess.py]: Changed it

File: requirements.txt
Comment (Flippchen) [reply requirements.txt:1]: Added them

**Review by anwielts — APPROVED** [review]
LGTM

**Pull request merged**